### PR TITLE
Move AddEntryToTree to shared source

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #495

### CODE REVIEW

File: shared/Microsoft.AspNetCore.Routing.UrlMatchingTree.Sources/TreeEnumerator.cs [hunk 1]
```diff
@@ -114,6 +118,194 @@ public void Reset()
             _stack.Clear();
             Current = null;
         }
-    }
 
+        internal static void AddEntryToTree(UrlMatchingTree tree, InboundRouteEntry entry)
```
Comment (rynowak) [inline shared/Microsoft.AspNetCore.Routing.UrlMatchingTree.Sources/TreeEnumerator.cs]: This should move to the `UrlMatchingTree` class. This isn't the right home for it.

**Comment by jbagga** [comment]
@rynowak Moved it to `UrlMatchingTree`

**Comment by jbagga** [comment]
This builds locally but not on the CI. I'll merge it after I know why

**Comment by pranavkm** [comment]
@jbagga see https://github.com/aspnet/Mvc/pull/7069 for the Travis failure.

**Comment by jbagga** [comment]
@pranavkm Okay thanks. Is it safe to merge this then?

**Comment by jbagga** [comment]
Looks like I should make the change to use the previous version of Appveyor to this repo too. I'll merge this after that change

**Comment by pranavkm** [comment]
Yeah, should be fine. I'd independently make the change to appv.yml to ensure future builds work.

**Pull request merged**